cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -373,7 +373,7 @@ func ManifestFilesAreEqual(path1, path2 string) (bool, string, error) {
 		return false, "", err
 	}
 
-	hasher := md5.New()
+	hasher := fnv.New128a()
```

Comment:
i think this may be fine becase it will result in a forced manifest upgrade for the release where this change is present.

but we'd have to run upgrade tests on this PR, to make sure.

cc @SataQiu @pacoxu @carlory @HirazawaUi

Reply:
this hash isn't exposed to users or stored to disk AFAICT, it's just used to do a deep comparison of these two pods by comparing their hashes?

Reply:
sha256 would be fine, but I am trying to follow the general guideline of:
- fnv-1a for non-cryptographic hashes (fast, and signals that it's not for security)
- sha256 for secure hashes (still fast, signals that it's for security)

fnv is currently used in the project. fnv1-a in 128bit is roughly equivalent to md5 while not being outdated or ambiguously for "security"

We could also look at comparing the pods some other way.

---

AFAICT this should not force any manifest upgrade, because the pods should still compare the same. We will hash both using fnv instead of md5

Reply:
ok, seems like fnv-1a 128 bit will be quite difficult to have collusions and we never got collusions with the md5 either.

/lgtm

Reply:
Yeah, we actually even use 32 and 64 bit fnv in other areas currently but I didn't think micro-optimizing made sense here and 128 bit ~matches md5 while being explicitly non-cryptographic and current. Initially I had sha256 but it was inconsistent with the larger PR as I worked through the remainder.